radish/documentors/__main__.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+"""
+This module provides a script to document Radish step implementations
+"""
+
+from sys import exit
+
+from os.path import split
+
+from docopt import docopt
+
+from radish import __VERSION__
+
+from radish.documentors.sphinx.multipage import MultiPageSphinxDocumentSet
+from radish.documentors.sphinx.singlepage import SinglePageSphinxDocument
+from radish.documentors.steps_writer import StepsDocumentWriter
+
+from radish.errororacle import error_oracle, enable_excepthook
+
+from radish.exceptions import RadishError
+
+from radish.loader import load_module_assets
+
+from radish.main import setup_config
+
+from radish.stepregistry import StepRegistry
+
+
+# ........................................................................... #
+@error_oracle
+def main():
+    usage = """
+Usage:
+    radish-docs sphinx <module-path>
+        [-o=<folder-path> | --output-folder=<folder-path>]
+        [--index-file-name=<file-path> ]
+        [-s=<file-path> | --single-file=<file-path>]
+        [-c | --print-to-console]
+        [--label-prefix=<label-prefix>]
+        [-r | --recreate]
+        [--no-ansi]
+    radish-docs (-h | --help)
+    radish-docs (-v | --version)
+
+Commands:
+    sphinx
+      create Sphinx documentation
+
+Arguments:
+    modules-path
+      path to python step modules to document
+
+Options:
+    -o=<folder-path>, --output-folder=<folder-path>
+      location of the output folder, by default `_build` in the current working
+      folder
+
+    --index-file-name=<file-name>
+      optional, name for the index file created by `--output-folder`, by
+      default "index.rst". This file is placed in the output folder.
+
+    -s=<file-path>, --single-file=<file-path>
+      write documentation to a single file
+
+    -c, --print-to-console
+      print `--single-file` documentaiton to console
+
+    --label-prefix=<label-prefix>
+      label prefix used for various section of the generated docs. Can be used
+      if you need a unique on in your documentation. By default set to
+      `radish.document_id.`
+
+    -r, --recreate
+      recreate output folder or file by removing and creating it again.
+
+    -h, --help
+      show this screen
+
+    -v, --version
+      show version
+
+(C) Copyright by Timo Furrer <tuxtimo@gmail.com>
+    """
+
+    # enable errororacles except hook system wide
+    enable_excepthook()
+
+    # parse sys.argv against our docopt argments
+    arguments = docopt("radish-docs {0}\n{1}".format(__VERSION__, usage),
+                       version=__VERSION__)
+
+    # create world wide configuration
+    setup_config(arguments)
+
+    # process arguments for "sphinx" command
+    if arguments['sphinx']:
```

Comment:
as far as I can tell, the else never reached since the commands are mandatory. It is also unnecessary to print docopt usage since docopt does it itself prints usage if a command like sphinx is not specified